Patient Registry System/PatientRegistrySystem.API/Controllers/AdminController.cs

```diff
@@ -0,0 +1,119 @@
+﻿using AutoMapper;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using PatientRegistrySystem.API.ViewModel;
+using PatientRegistrySystem.API.ViewModel.Registration;
+using PatientRegistrySystem.DB.Entities;
+using PatientRegistrySystem.Domain.Dto;
+using PatientRegistrySystem.Services;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace PatientRegistrySystem.API.Controllers
+{
+    [ApiController]
+    [Route("[controller]")]
+    [Authorize(Roles = "Admin")]
+    public class AdminController : Controller
+    {
+        private readonly UserManager<ApplicationUser> _userManager;
+        private readonly IMapper _mapper;
+        private readonly IUserService _userService;
+
+        public AdminController(UserManager<ApplicationUser> userManager,
+             IMapper mapper,
+            IUserService userService)
+        {
+            _userManager = userManager;
+            _mapper = mapper;
+            _userService = userService;
+        }
+
+        [Route("[action]")]
+        [HttpPost]
+        [Authorize(Roles = "Admin")]
+        public async Task<IActionResult> CreateDoctor([FromBody] RegistrationDoctoreViewModel model)
+        {
+            if (ModelState.IsValid)
+            {
+                ApplicationUser applicationUser = new ApplicationUser()
+                {
+                    Email = model.Email,
+                    UserName = model.Email,
+                    PhoneNumber = model.Phone,
+                    User = _mapper.Map<User>(model)
+                };
+
+                applicationUser.User.ApplicationUserId = applicationUser.Id;
+                applicationUser.User.ApplicationUser = applicationUser;
+                applicationUser.User.Doctor.Add(new Doctor()
+                {
+                    Address1 = model.Adress1,
+                    Address2 = model.Adress2,
+                    UserId = applicationUser.User.UserId,
+                    User = applicationUser.User
+                });
+
+                var result = await _userManager.CreateAsync(applicationUser, model.Password);
```

Comment:
This logic should exists in the controller, you should move it to an DoctorService. Same as you did with the UserService.

Reply:
Same thing applies to all the other logic you have in the APIs in this class.